Makefile

```diff
@@ -1,13 +1,42 @@
 SHELLCHECK_CMD := shellcheck bash/scripts/*
 BATS_CMD := bats --tap bash/tests
+
 TEST_ENV_PREFIX := docker run --rm -v ${CURDIR}:/workdir -w /workdir quay.io/deis/shell-dev
+GRADLE_TEST_CMD := docker run --name gradle-test -v ${CURDIR}:/workdir -w /workdir frekele/gradle:3.0-jdk8 ./gradlew test
 
-test:
+test-style:
 	${SHELLCHECK_CMD}
+
+test-scripts:
 	${BATS_CMD}
 
-docker-test:
+test-dsl:
+	./gradlew test
+
+test: test-style test-scripts test-dsl
+
+docker-test-style:
 	${TEST_ENV_PREFIX} ${SHELLCHECK_CMD}
+
+docker-test-scripts:
 	${TEST_ENV_PREFIX} ${BATS_CMD}
 
-.PHONY: test docker-test
+docker-test-dsl:
+	docker rm gradle-test || true
```

Comment:
In makefiles, you want to prefix a line with `-` if you don't care about the error: https://www.gnu.org/software/make/manual/html_node/Errors.html